qemu/tests/dump_guest_memory.py

```diff
@@ -0,0 +1,198 @@
+
+    def check_result(qmp_o, expect_o=None):
+        """
+        Check test result with difference way according to result_check.
+        result_check = equal, expect_o should equal to qmp_o.
+        result_check = contain, expect_o should be contained in qmp_o
+
+        :param qmp_o: output from qmp_cmd.
+        :type qmp_o: list
+        :param expect_o: the expect result.
+        :type expect_o: dict
+
+        :return check result
+        :rtype: bool
+        """
+        logging.info("Expect result is %s" % expect_o)
+        logging.info("Actual result that get from qmp_cmd is %s" % qmp_o)
+        result = None
+        if result_check == "equal":
+            if not operator.eq(qmp_o, expect_o):
+                test.fail("QMP output does not equal to the expect result.\n "
+                          "Expect result: '%s'\n"
+                          "Actual result: '%s'" % (expect_o, qmp_o))
+        elif result_check == "contain":
+            if len(expect_o) == 0:
+                result = True
+            elif isinstance(expect_o, dict):
+                for key, val in expect_o.items():
+                    result = check_list(qmp_o, key, val)
+        return result
+
+    def execute_qmp_cmd(qmp_cmd, expect_result):
+        """
+        Execute qmp command and check if result as expect
+
+        :param qmp_cmd: qmp command
+        :type qmp_cmd: str
+        :param expect_result: expect result of qmp command
+        :type expect_result: str
+
+        :return check result
+        :rtype: bool
+        """
+        # qmp command
+        try:
+            # Testing command
+            logging.info("Run qmp command '%s'.", qmp_cmd)
+            qmp_o = qmp_port.send_args_cmd(qmp_cmd)
+            logging.debug("QMP command:'%s' \n Output: '%s'", qmp_cmd, [qmp_o])
+        except Exception as err:
+            qmp_o = err.data
+            logging.info(err)
+
+        if result_check:
+            logging.info("Verify qmp command '%s'." % qmp_cmd)
+            return check_result([qmp_o], eval(expect_result))
+
+    def check_dump_file():
+        """
+        Use crash to check dump file
+        """
+        process.run("echo bt > %s" % crash_script)
+        process.run("echo quit >> %s" % crash_script)
+        crash_cmd = "crash -i %s /usr/lib/debug/lib/modules/%s/vmlinux "
+        crash_cmd %= (crash_script, host_kernel_version)
+        crash_cmd += dump_file
+        status, output = process.getstatusoutput(crash_cmd)
+        os.remove(crash_script)
+        logging.debug(output)
+        if status != 0:
+            test.fail("vmcore corrupt")
+        elif output.find("error") >= 0:
+            test.fail("vmcore corrupt")
+
+    # install crash/gdb/kernel-debuginfo in host
+    utils_package.package_install(["crash", "gdb", "kernel-debuginfo*"])
+
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+    host_kernel_version = (process.getoutput("uname -r")).strip()
```

Comment:
`host_kernel_version = process.getoutput("uname -r").strip()` would be fine.

Reply:
Thanks for pointing this out, it has been updated.